### ISSUE TITLE
Windows builds of etcd 3.x via build.bat throw spurious warnings / queries while being hard to parse and debug

### ISSUE THREAD

**Issue by MartyMacGyver** [coreos/etcd#5866]
Such spurious warnings and unnecessary interactions should be eliminated. In addition, the output should be more informative.

Note: build.bat is not the problem here - the PowerShell script build.ps1 needs updating.

A clean build of etcd produces the following output (etcd 3.0.0):

    G:\go\src\github.com\coreos\etcd>build.bat
    
    G:\go\src\github.com\coreos\etcd>powershell -ExecutionPolicy Bypass -File build.ps1
    Rebuilding symlinks
    The directory name is invalid.
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\etcdctl <<===>> G:\go\src\github.com\coreos\etcd\etcdctl
    The directory name is invalid.
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\etcdmain <<===>> G:\go\src\github.com\coreos\etcd\etcdmain
    Hardlink created for G:\go\src\github.com\coreos\etcd\cmd\main.go <<===>> G:\go\src\github.com\coreos\etcd\main.go
    The directory name is invalid.
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\tools <<===>> G:\go\src\github.com\coreos\etcd\tools
    The directory name is invalid.
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\vendor\github.com\coreos\etcd <<===>> G:\go\src\github.com\coreos\etcd\
    The system cannot find the path specified.
    The system cannot find the path specified.
    Junction created for G:\go\src\github.com\coreos\etcd\gopath\src\github.com\coreos\etcd <<===>> G:\go\src\github.com\coreos\etcd

Rebuilds get even more confusing, with ambiguous user interaction required:

    G:\go\src\github.com\coreos\etcd>build.bat
    
    G:\go\src\github.com\coreos\etcd>powershell -ExecutionPolicy Bypass -File build.ps1
    Rebuilding symlinks
    Could Not Find G:\go\src\github.com\coreos\etcd\cmd\etcdctl
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\etcdctl <<===>> G:\go\src\github.com\coreos\etcd\etcdctl
    Could Not Find G:\go\src\github.com\coreos\etcd\cmd\etcdmain
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\etcdmain <<===>> G:\go\src\github.com\coreos\etcd\etcdmain
    Hardlink created for G:\go\src\github.com\coreos\etcd\cmd\main.go <<===>> G:\go\src\github.com\coreos\etcd\main.go
    Could Not Find G:\go\src\github.com\coreos\etcd\cmd\tools
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\tools <<===>> G:\go\src\github.com\coreos\etcd\tools
    Could Not Find G:\go\src\github.com\coreos\etcd\cmd\vendor\github.com\coreos\etcd
    Junction created for G:\go\src\github.com\coreos\etcd\cmd\vendor\github.com\coreos\etcd <<===>> G:\go\src\github.com\coreos\etcd\
    G:\go\src\github.com\coreos\etcd\gopath\src\github.com\coreos\*, Are you sure (Y/N)? Y
    A subdirectory or file G:\go\src\github.com\coreos\etcd\gopath\src\github.com\coreos already exists.
    Junction created for G:\go\src\github.com\coreos\etcd\gopath\src\github.com\coreos\etcd <<===>> G:\go\src\github.com\coreos\etcd

**Comment by MartyMacGyver** [comment]
Attached is a tested example of how this might be done, eliminating unwanted warnings and queries while adding appropriate informational output:

[build.ps1.txt](https://github.com/coreos/etcd/files/345989/build.ps1.txt)

**Comment by xiang90** [comment]
@sinsharat Can you check if this is still a problem for windows?

**Comment by sinsharat** [comment]
@xiang90 sure once i finish the etcd runner code changes will look into this.

**Comment by MartyMacGyver** [comment]
I just cloned, built and rebuilt etcd as I described above... same results.

I could create and submit a patch tomorrow that fixes the issues I'm seeing.

**Comment by xiang90** [comment]
@MartyMacGyver Sure. That would be awesome!
